### PR TITLE
Implement read_csv in cudf-polars using pylibcudf

### PR DESCRIPTION
**Pull request by lithomas1**
## Description

Replace cudf-classic with pylibcudf for CSV reading in cudf-polars

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf_polars/tests/test_scan.py [hunk 1]
```diff
@@ -129,6 +130,30 @@ def test_scan_csv_column_renames_projection_schema(tmp_path):
...
+def test_scan_csv_multi_differing_colnames(tmp_path):
+    with (tmp_path / "test1.csv").open("w") as f:
+        f.write("""foo,bar,baz\n1,2\n3,4,5""")
+    with (tmp_path / "test2.csv").open("w") as f:
+        f.write("""abc,def,ghi\n1,2\n3,4,5""")
+    q = pl.scan_csv(
+        [tmp_path / "test1.csv", tmp_path / "test2.csv"],
+    )
+    with pytest.raises(pl.exceptions.ComputeError):
+        q.collect(post_opt_callback=execute_with_cudf)
```
Comment (wence-) [inline python/cudf_polars/tests/test_scan.py:154]: What is this testing?

**Review by wence- — CHANGES_REQUESTED** [review]
Small changes

File: python/cudf_polars/tests/test_scan.py (hunk 1 above)
Comment (lithomas1) [reply python/cudf_polars/tests/test_scan.py:154]: This is testing cases where the column names are different between CSV files in the multiple CSV file case.

It looks like polars checks this for us (but it's good to make sure that's the case with a test).

File: python/cudf_polars/cudf_polars/dsl/ir.py [hunk 2]
```diff
@@ -278,33 +274,39 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
 
             # polars skips blank lines at the beginning of the file
             pieces = []
+            colnames = None
             for p in self.paths:
                 skiprows = self.reader_options["skip_rows"]
-                # TODO: read_csv expands globs which we should not do,
-                # because polars will already have handled them.
```
[the remark this answers is not in the document]
Comment (lithomas1) [reply python/cudf_polars/cudf_polars/dsl/ir.py:284]: Added.

File: python/cudf_polars/cudf_polars/dsl/ir.py [hunk 3]
```diff
@@ -278,33 +274,40 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
 
             # polars skips blank lines at the beginning of the file
             pieces = []
+            colnames = None
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/ir.py:277]:
```suggestion
```

File: python/cudf_polars/cudf_polars/dsl/ir.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wence-) [reply python/cudf_polars/cudf_polars/dsl/ir.py:284]: I don't see it. What we want to check is that (with `test1.csv`, `test2.csv`, `test*.csv`)

```
pl.scan_csv("test*.csv", glob=False)
```
reads the single `test*.csv` file correctly.

**Review by wence- — APPROVED** [review]
Nearly there, some small issues to address but approving on the assumption that will be done. Thanks!

File: python/cudf_polars/cudf_polars/dsl/ir.py [hunk 4]
```diff
@@ -278,33 +274,40 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
...
+                    na_filter=True,
+                    na_values=null_values,
+                    keep_default_na=False,
+                    skiprows=skiprows,
+                    comment=comment,
+                    decimal=decimal,
+                    dtypes=self.schema,
+                )
+                pieces.append(tbl_w_meta)
+            tbls, colnames = zip(
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/ir.py:301]:
```suggestion
            tables, colnames = zip(
```
We're not writing the SYS-V operating system.

File: python/cudf_polars/cudf_polars/dsl/ir.py (hunk 3 above)
Comment (lithomas1) [reply python/cudf_polars/cudf_polars/dsl/ir.py]: Thanks.

File: python/cudf_polars/cudf_polars/dsl/ir.py (hunk 4 above)
Comment (lithomas1) [reply python/cudf_polars/cudf_polars/dsl/ir.py]: 😅

File: python/cudf_polars/cudf_polars/dsl/ir.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lithomas1) [reply python/cudf_polars/cudf_polars/dsl/ir.py:284]: Oh, I completely misunderstood there. 

I added a glob=False test case in the latest commit.

**Pull request merged**